### PR TITLE
✨ Gulp task to convert analytic vendor JS configs to JSON

### PR DESCRIPTION
**Pull request by jonathantyng-amp**
Adds a gulp task to convert the analytic vendor JS files to JSON files. 

cc @zhouyx for review

### CODE REVIEW

**Comment by jonathantyng-amp** [comment]
cc @zhouyx for review!

File: build-system/tasks/generate-vendor-jsons.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  const srcDir = 'extensions/amp-analytics/0.1';
+  const srcFile = 'vendors.js';
+  const tempPath = 'dist/temp-analytics/';
+  const destPath = 'extensions/amp-analytics/0.1/vendors/';
+  const compileOptions = {
+    browserifyOptions: {
+      // allow gulp to import this file after compile
+      standalone: 'analyticsVendors',
+      // need to stub self since we are not running this in the browser
+      insertGlobalVars: {
```
Comment (zhouyx) [inline build-system/tasks/generate-vendor-jsons.js]: Looks very hacky, but we should get rid of this very soon. So I'm fine with it. 

@erwinmombay @rsimha to make the final call.

**Comment by rsimha** [comment]
@jonathantyng-amp, before I do a full review, could you help me understand what this task does, how long it takes, and when you typically run it? (Development? Release? What about minified vs. unminified builds?)

Our browserify configs are currently fragmented (I'm in the process of trying to consolidate it), so I'd like to avoid further fragmentation if possible.

**Comment by jonathantyng-amp** [comment]
@rsimha 
Context: `amp-analytics` uses ~70 vendor configs that are currently all compiled into the bundle. We want to separate these out to lazy load them instead. As part of this process we want to convert each vendor config into its own JSON file.

This gulp task converts the current vendor configs into their own JSON file, placing the files into `/extensions/amp-analytics/0.1/vendors/`. This gulp task will only be run in the next several weeks temporarily while we run some AB tests to make sure the vendor configs stay in sync. After the AB tests are complete, we will permanently commit the JSON files and this gulp task will not be needed anymore.

This task will be run in the release process and runs in ~3 sec

**Comment by zhouyx** [comment]
🎉 Merged

**Pull request merged**